reactive/containerd.py

```diff
@@ -193,6 +195,41 @@ def config_changed():
         status_set('blocked', 'Container runtime not available.')
 
 
+@when_any('config.changed.http_proxy', 'config.changed.https_proxy',
+          'config.changed.no_proxy')
+@when('containerd.ready')
+def proxy_changed():
+    """
+    Apply new proxy settings.
+
+    :returns: None
+    """
+    context = config()
+    service_file = 'proxy.conf'
+    service_directory = '/etc/systemd/system/containerd.service.d'
+    service_path = os.path.join(service_directory, service_file)
+
+    if context.get('http_proxy') or \
+            context.get('https_proxy') or context.get('no_proxy'):
+
+        os.makedirs(service_directory, exist_ok=True)
+
+        render(
+            service_file,
+            service_path,
+            context
+        )
+
+    else:
+        try:
+            os.remove(service_path)
+        except FileNotFoundError:
+            return  # We don't need to restart the daemon.
+
+    check_call(['systemctl', 'daemon-reload'])
+    host.service_restart('containerd.service')
```

Comment:
What happens here in the case where you just removed, or never created, the service file?

Reply:
See the comment below re. service directories vs unit files.  If the inherited file no longer exists, it is ignored.